// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  
  //Vote Request and Response
  rpc AskForVote (VoteRequest) returns (VoteReply) {}
  
  // Map function
  rpc Mapper(MapRequest) returns (MapReply) {}
  
  // Reduce function
  rpc Reducer(ReduceRequest) returns (ReduceReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
  string nodeid = 2;
  string nodestatus = 3;
  string termnum = 4;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message VoteRequest {
  string name = 1;
  string termnum = 2;
}

message VoteReply {
  string message = 1;
}

// The request message containing the input for the map function
message MapRequest {
  string name = 1;
}

// The response message containing the output of the map function
message MapReply {
  string message = 1;
}

// The request message containing the input for the reduce function
message ReduceRequest {
  string name = 1;
}

// The response message containing the output of the reduce function
message ReduceReply {
  string message = 1;
}
